src/components/AddPlaidBankAccount.tsx

```diff
@@ -24,103 +23,82 @@ import PlaidLink from './PlaidLink';
 import RadioButtons from './RadioButtons';
 import Text from './Text';
 
-const propTypes = {
+type AddPlaidBankAccountOnyxProps = {
     /** If the user has been throttled from Plaid */
-    isPlaidDisabled: PropTypes.bool,
+    isPlaidDisabled: OnyxEntry<boolean>;
 
+    /** Plaid SDK token to use to initialize the widget */
+    plaidLinkToken: OnyxEntry<string>;
+};
+
+type AddPlaidBankAccountProps = AddPlaidBankAccountOnyxProps & {
     /** Contains plaid data */
-    plaidData: plaidDataPropTypes.isRequired,
+    plaidData: OnyxEntry<PlaidData>;
 
     /** Selected account ID from the Picker associated with the end of the Plaid flow */
-    selectedPlaidAccountID: PropTypes.string,
-
-    /** Plaid SDK token to use to initialize the widget */
-    plaidLinkToken: PropTypes.string,
+    selectedPlaidAccountID?: string;
 
     /** Fired when the user exits the Plaid flow */
-    onExitPlaid: PropTypes.func,
+    onExitPlaid?: () => void;
 
     /** Fired when the user selects an account */
-    onSelect: PropTypes.func,
+    onSelect?: (plaidAccountID: string) => void;
 
     /** Additional text to display */
-    text: PropTypes.string,
+    text?: string;
 
     /** The OAuth URI + stateID needed to re-initialize the PlaidLink after the user logs into their bank */
-    receivedRedirectURI: PropTypes.string,
+    receivedRedirectURI: string | null;
 
     /** During the OAuth flow we need to use the plaidLink token that we initially connected with */
-    plaidLinkOAuthToken: PropTypes.string,
+    plaidLinkOAuthToken?: string;
 
     /** If we're updating an existing bank account, what's its bank account ID? */
-    bankAccountID: PropTypes.number,
+    bankAccountID?: number;
 
     /** Are we adding a withdrawal account? */
-    allowDebit: PropTypes.bool,
+    allowDebit?: boolean;
 
     /** Is displayed in new VBBA */
-    isDisplayedInNewVBBA: PropTypes.bool,
+    isDisplayedInNewVBBA?: boolean;
 
     /** Text to display on error message */
-    errorText: PropTypes.string,
+    errorText?: string;
 
     /** Function called whenever radio button value changes */
-    onInputChange: PropTypes.func,
-};
-
-const defaultProps = {
-    selectedPlaidAccountID: '',
-    plaidLinkToken: '',
-    onExitPlaid: () => {},
-    onSelect: () => {},
-    text: '',
-    receivedRedirectURI: null,
-    plaidLinkOAuthToken: '',
-    allowDebit: false,
-    bankAccountID: 0,
-    isPlaidDisabled: false,
-    isDisplayedInNewVBBA: false,
-    errorText: '',
-    onInputChange: () => {},
+    onInputChange?: (plaidAccountID: string) => void;
 };
 
 function AddPlaidBankAccount({
     plaidData,
-    selectedPlaidAccountID,
+    selectedPlaidAccountID = '',
     plaidLinkToken,
-    onExitPlaid,
-    onSelect,
-    text,
-    receivedRedirectURI,
-    plaidLinkOAuthToken,
-    bankAccountID,
-    allowDebit,
+    onExitPlaid = () => {},
+    onSelect = () => {},
```

Comment:
I think you can remove the default `() => {}` values and handle this in the code using optional chaining, wdyt?

Reply:
@fabioh8010  onSelect cannot be undefined when passed to Picker 😅 I think it's safe to leave it as it is